src/primitives/transaction.h

```diff
@@ -50,6 +50,8 @@ class COutPoint
     std::string ToString() const;
 };
 
+std::size_t hash_value(const COutPoint& o);
```

Comment:
Yep. We'll proceed with a `std::map`, since the performance difference to a `boost::unordered_map` are not noticeable.